src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -341,170 +603,333 @@ function saturation_shum(T, ρ, q_liq=0, q_ice=0)
 
     # liquid-ice potential temperature, approximating latent heats
     # of phase transitions as constants
-    return dry_pottemp(T, p, q_tot, q_liq, q_ice) * exp(-(LH_v0*q_liq + LH_s0*q_ice)/(_cp_m*T))
-
+    return dry_pottemp(T, p, q_tot, q_liq, q_ice) * exp(-(DT(LH_v0)*q_liq + DT(LH_s0)*q_ice)/(_cp_m*T))
 end
 
 """
-    dry_pottemp(T, p, q_tot=0, q_liq=0, q_ice=0)
+    liquid_ice_pottemp(ts::ThermodynamicState)
 
-Return the dry potential temperature, given the temperature `T`,
-pressure `p`, total specific humidity `q_tot`, liquid specific humidity `q_liq`,
-and ice specific humidity `q_ice`.
+The liquid-ice potential temperature,
+given a thermodynamic state `ts`.
 """
-dry_pottemp(T, p, q_tot=0, q_liq=0, q_ice=0) = T / exner(p, q_tot, q_liq, q_ice)
+liquid_ice_pottemp(ts::ThermodynamicState) =
+  liquid_ice_pottemp(air_temperature(ts), air_pressure(ts), ts.q_tot,
+                     phase_partitioning_eq(ts)...)
 
 """
-    exner(p, q_tot=0, q_liq=0, q_ice=0)
+    dry_pottemp(T, p, q_tot, q_liq, q_ice)
 
-Return the Exner function, given the pressure `p`, total specific
-humidity `q_tot`, liquid specific humidity `q_liq`, and ice specific humidity `q_ice`.
+The dry potential temperature where
+
+ - `T` temperature
+ - `p` pressure
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+ """
+dry_pottemp(T::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT} =
+  T / exner(p, q_tot, q_liq, q_ice)
+
+"""
+    dry_pottemp(ts::ThermodynamicState)
+
+The dry potential temperature, given a thermodynamic state `ts`.
 """
-function exner(p, q_tot=0, q_liq=0, q_ice=0)
+dry_pottemp(ts::ThermodynamicState) =
+  dry_pottemp(air_temperature(ts), air_pressure(ts), ts.q_tot,
+              phase_partitioning_eq(ts)...)
+
 
+"""
+    virtual_temp(T, p, q_tot, q_liq, q_ice)
+
+The virtual temperature where
+
+ - `T` temperature
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+virtual_temp(T::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT} =
+  gas_constant_air(q_tot, q_liq, q_ice)/R_d*dry_pottemp(T, p, q_tot, q_liq, q_ice)
+
+"""
+    virtual_temp(ts::ThermodynamicState)
+
+The virtual potential temperature,
+given a thermodynamic state `ts`.
+"""
+virtual_temp(ts::ThermodynamicState) =
+  virtual_temp(air_temperature(ts), ts.q_tot, phase_partitioning_eq(ts)...)
+
+
+"""
+    liquid_ice_pottemp_sat(T, p, q_tot, q_liq, q_ice)
+
+The saturated liquid ice potential temperature where
+
+ - `T` temperature
+ - `p` pressure
+and, optionally,
+ - `q_tot` total specific humidity
+ - `q_liq` liquid specific humidity
+ - `q_ice` ice specific humidity
+"""
+function liquid_ice_pottemp_sat(T::DT, p::DT, q_tot::DT=DT(0), q_liq::DT=DT(0), q_ice::DT=DT(0)) where {DT}
+    ρ = air_density(T, p, q_tot, q_liq, q_ice)
+    q_vs = saturation_shum(T, ρ, q_liq, q_ice)
```

Comment:
We should use a uniform variable for saturation specific humidity, e.g., 'q_v_sat' would be consistent with our other notation. (Above the same variable is `q_vstar`.)